src/App.tsx

```diff
@@ -1,223 +1,162 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
-import { AuthContext } from './components/Auth/AuthContext';
+// eslint-disable-next-line object-curly-newline
+import React, { useEffect, useRef, useState } from 'react';
+// eslint-disable-next-line object-curly-newline
+import { getTodos, createTodo, deleteTodo, updateTodo } from './api/todos';
+import { filterTotos } from './api/filter';
+import { useAuthContext } from './components/Auth/useAuthContext';
+// eslint-disable-next-line object-curly-newline
+import { ErrorNotification, Header, TodoList, Footer } from './components';
+import { Todo, FilterTypes, ErrorTypes } from './types';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const user = useContext(AuthContext);
+  const user = useAuthContext();
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [typeFilter, setTypeFilter] = useState(FilterTypes.All);
+  const [error, setError] = useState('');
+  const [isHiddenErrorNote, setIsHiddenErrorNote] = useState(true);
```

Comment:
It is difficult logic. you have hidden error by default, so it's better to name variable isErrorNoteShown and set it as false. And only when you need to show the error you do setErrorNoteShown(true) - so now it really do what it suppose to do